exec/vector/src/main/java/org/apache/drill/exec/vector/VectorUtils.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.vector;
+
+public class VectorUtils {
+
+  /**
+   * Vectors cannot be any larger than the Netty memory allocation
+   * block size.
+   */
+
+  private static final int ABSOLUTE_MAX_SIZE = 16 * 1024 * 1024;
+  private static final int ABSOLUTE_MIN_SIZE = 16 * 1024;
```

Comment:
This code allows a boot-time option to limit maximum vector size. I thought it would be handy for testing, but turned out not to be necessary, so I might remove it.

To prevent failures when the value is set wrong, I clamped the value to some minimum and maximum. The maximum is set by the Netty block size. The minimum I just made up: seems if we limit vector size to much smaller than 16 K we'd be unlikely to make progress.

Added a comment to clarify.